ansible/kubernetes/README.md

```diff
@@ -5,97 +5,150 @@ Environment.
-- Configures the firewall for use with Kubernetes.
-- Installs and starts docker.
-- Installs kubeadm, kubectl, and kubelet.
-- Starts kubelet.
+- Installs wireguard on the nodes for use with the wireguard-native CSI driver.
+- Configures the firewall for use with K3s.
+- Sets up support for pulling images from the CyVerse image repo.
+- Installs and runs the installation script for k3s.
+- Joins nodes to the cluster.
+- Labels and taints nodes for running DE services, VICE apps, and GPU-enabled VICE-apps.
+- Installs the IRODS CSI driver.
+- Sets up namespaces, service accounts, roles, role bindings, and cluster role bindings.
 
-### create.yml
+### install_csi_driver.yml
 
-This playbook creates the Kubernetes cluster:
+Installs the CSI driver into the k3s cluster. Also runs as part of the `install_k3s.yml` playbook.
 
-- Initializes the cluster on the first node listed in the `k8s-controllers` group.
-- Joins the remaning nodes in the `k8s_controllers` group.
-- Joins the nodes in the `k8s_workers` group, which is comprised of the `k8s_de_workers`, `k8s_vice_workers`, and
-  `k8s_gpu_workers` groups.
-- Labels the nodes in the `k8s_de_workers` group.
-- Taints and labels the nodes in the `k8s_vice_workers` group.
-- Taints and labels the nodes in the `k8s_gpu_workers` group.
+### setup_haproxy.yml
+
+Sets up the HAProxy in front of the K3s controllers. Runs as part of the `install_k3s.yml` playbook. Shouldn't need to be run manually, but it's here just in case.
+
+### uninstall_k8s.yml
+
+Uninstalls the current k8s cluster. Be very careful with this one, likely won't need to be run more than once internally and never for external deployments. Will probably be removed at some point in the future.
+
+### update.yml
+
+Runs updates on the k3s/k8s nodes. Likely will be moved elsewhere in the future.
 
 ## Inventory Setup
 
 The inventory for these playbooks is relatively simple:
 
 ```
-[k8s:children]
-k8s_nodes
-k8s_api_proxy
-
-[k8s_nodes:children]
-k8s_controllers
-k8s_workers
-
-[k8s_workers:children]
-k8s_de_workers
-k8s_vice_workers
-k8s_gpu_workers
-
-[k8s_api_proxy]
-k8s-haproxy.example.org
-
-[k8s_controllers]
-k8s-con-1.example.org
-k8s-con-2.example.org
-k8s-con-3.example.org
-
-[k8s_de_workers]
-k8s-worker-1.example.org
-k8s-worker-2.example.org
-k8s-worker-3.example.org
-k8s-worker-4.example.org
-k8s-worker-5.example.org
-k8s-worker-6.example.org
-k8s-worker-7.example.org
-
-[k8s_vice_workers]
-k8s-vice-1.example.org
-k8s-vice-2.example.org
-k8s-vice-3.example.org
-k8s-vice-4.example.org
-
-[k8s_gpu_workers]
-k8s-gpu-1.example.org
-k8s-gpu-2.example.org
-k8s-gpu-3.example.org
-k8s-gpu-4.example.org
+[k3s:children]
+k3s_nodes
+k3s_api_proxy
+
+[k3s_nodes:children]
+k3s_controllers
+k3s_workers
+
+[k3s_workers:children]
+k3s_de_workers
+k3s_vice_workers
+k3s_gpu_workers
+
+[k3s_api_proxy]
+k3s-haproxy.example.org
+
+[k3s_controllers]
+k3s-con-1.example.org
+k3s-con-2.example.org
+k3s-con-3.example.org
+
+[k3s_de_workers]
```

Comment:
If we're changing this here, we should change it in the haproxy playbooks (for the DE haproxy) as well. Unless that playbook shouldn't use these groups, but it seems like unnecessary duplication not to.

Reply:
Good catch, thanks. I'll look into getting it changed as a part of this PR.

Reply:
Should be done now, let me know if I missed anything else.